integration-tests/oidc-code-flow/src/test/java/io/quarkus/it/keycloak/CodeFlowTest.java

```diff
@@ -111,6 +118,67 @@ public void testTokenTimeoutLogout() throws IOException, InterruptedException {
         }
     }
 
+    @Test
+    public void testRPInitiatedLogout() throws IOException, InterruptedException {
+        Keycloak keycloak = KeycloakRealmResourceManager.createKeycloakClient();
+
+        RealmResource realm = keycloak.realm(KeycloakRealmResourceManager.KEYCLOAK_REALM);
+        RealmRepresentation representation = realm
+                .toRepresentation();
+
+        Integer ssoSessionMaxLifespan = representation.getSsoSessionMaxLifespan();
+        representation.setSsoSessionMaxLifespan(20);
+
+        Integer accessCodeLifespan = representation.getAccessCodeLifespan();
+        representation.setAccessTokenLifespan(5);
+
+        realm.update(representation);
+
+        try (final WebClient webClient = createWebClient()) {
+            HtmlPage page = webClient.getPage("http://localhost:8081/tenant-logout");
+            assertEquals("Log in to quarkus", page.getTitleText());
+            HtmlForm loginForm = page.getForms().get(0);
+            loginForm.getInputByName("username").setValueAttribute("alice");
+            loginForm.getInputByName("password").setValueAttribute("alice");
+            page = loginForm.getInputByName("login").click();
+            assertTrue(page.asText().contains("Tenant Logout"));
+            assertNotNull(getSessionCookie(webClient));
+
+            page = webClient.getPage("http://localhost:8081/tenant-logout/logout");
+            assertTrue(page.asText().contains("You were logged out"));
+            assertNull(getSessionCookie(webClient));
+
+            page = webClient.getPage("http://localhost:8081/tenant-logout");
+            assertEquals("Log in to quarkus", page.getTitleText());
+            loginForm = page.getForms().get(0);
+            loginForm.getInputByName("username").setValueAttribute("alice");
+            loginForm.getInputByName("password").setValueAttribute("alice");
+            page = loginForm.getInputByName("login").click();
+            assertTrue(page.asText().contains("Tenant Logout"));
+
+            //wait now so that we reach the refresh timeout
+            await().atMost(10, TimeUnit.SECONDS)
+                    .pollInterval(Duration.ofSeconds(5))
+                    .until(new Callable<Boolean>() {
+                        @Override
+                        public Boolean call() throws Exception {
+                            webClient.getOptions().setRedirectEnabled(false);
+                            WebResponse webResponse = webClient
+                                    .loadWebResponse(new WebRequest(URI.create("http://localhost:8081/tenant-logout").toURL()));
+                            // Should not redirect to OP but silently refresh token
```

Comment:
What would be really good is to verify `idtoken` (and also AT at least) has been refreshed. There is a code somewhere in this class which checks the cookie content (disables the auto-redirect and then re-enables - not sure it will be needed yet). So in this test you can record `id_token` before the refresh and assert it is not the same after the refresh.

Also, this is not part of this PR, but there is a timeout test here as well, so please add awaitility there, sorry :-)

Reply:
I'm already using awaitility, no ?

Reply:
@pedroigor Sorry, there is an existing test there testTimeout I think which uses `wait(10)`.
And verifying the id token value is not the same would be good